enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,170 @@
+     alerts are firing.
+    - Provide option to turn off an enforcing mode
+
+### Non-Goals
+- Allow cluster admins to set the rules and externalize the configuration
+
+## Open Questions 
+
+1. What are the different ways in which a PDB & SCCs are deemed invalid by an operator action
+2. Configuration option on OLM to make for a permissive mode vs. a enforcing mode 
+ - How do we provide flexibility to cluster admin to define what rules are permitted vs enforced    
+3. Validation library is used to define what components of an Operator are deemed potentially harmful 
+	 - How do I go about finding this? Also, for these components that are deemed harmful, what is the expected action to be taken? I
+	 presume they would need a different design strategy than the one for SCCs and PDB? 
+
+## Design Details
+
+The design is split into 3 sections, each one per goal defined above. 
+
+## 1. A mechanism to alert when an invalid PDB is being applied on to a cluster (a.k.a validating PDB yaml) 
+
+Let's consider an example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+
+PodDisruptionBudget has three fields:
+
+- A label selector `.spec.selector` to specify the set of pods to which it applies. This field is required.
+- `.spec.minAvailable` which specifies the number of pods that must still be available after the eviction, even in the absence of the evicted pod.
+- `.spec.maxUnavailable` which specifies the number of pods that can be unavailable after the eviction
+
+Note: `minAvailable` & `maxUnavailable` can be either an absolute number or a percentage.
+
+An PDB is considered invalid if any of these fields has incorrect values and gets applied on to a cluster.
+
+Below lists a number of potential approaches:
+
+### 1. Add a command line flag (Straw Man proposal)
+The flag parses the PDB yaml and validates the fields 
+ 
+### 2. Introduce Admission webhooks
+- Kubernetes allows decoupling policy decisions from the API server by means of admission controller webhooks 
+to intercept admission requests before they are persisted as objects in Kubernetes
+- Have a set of predefined rules as part of webhooks to guard against harmful operator actions
+
+#### Pros
+- Provides flexibility by either offering to validate objects or mutate(either by setting default values/ update field) using
+validating or mutating webhooks respectively 
+
+ 
+### 3. [Needs more investigation] Using [CUE](https://cuelang.org/docs/integrations/yaml/#validate-yaml-files)
+- CUE is a configuration language that allows you to define a `<pdb-checks>.cue` template for the PDB yaml
+- CUE can be used as a go library to define the template and validate the PDBs before being applied on to the cluster   
+
+#### Pros
+-  
+
+#### Cons
+- Not a lot to offer in terms of flexibility as compared to OPA (discussed next)
```

Comment:
Another con could be that CUE is relatively new and is under heavy development so there's a high chance of our velocity being affected by bugs in CUE